src/Aspire.Hosting/Publishing/ResourceContainerImageBuilder.cs

```diff
@@ -273,19 +273,22 @@ private async Task<bool> ExecuteDotnetPublishAsync(IResource resource, Container
 
             if (options.TargetPlatform is not null)
             {
-                // Use the appropriate MSBuild property based on the number of RIDs
+                // Use the appropriate MSBuild properties based on the number of RIDs
                 var runtimeIds = options.TargetPlatform.Value.ToMSBuildRuntimeIdentifierString();
                 var ridArray = runtimeIds.Split(';');
 
                 if (ridArray.Length == 1)
                 {
-                    // Single platform - use ContainerRuntimeIdentifier
+                    // Single platform - use RuntimeIdentifier/ContainerRuntimeIdentifier
+                    arguments += $" /p:RuntimeIdentifier=\"{ridArray[0]}\"";
```

Comment:
Question: What was the failure mode here? From what I recall, we always just set `ContainerRuntimeIdentifier` since this was [introduced in 9.4](https://github.com/dotnet/aspire/blob/release/9.4/src/Aspire.Hosting/Publishing/ResourceContainerImageBuilder.cs#L278). Was this breaking AoT in some way?